src/algo.rs

```diff
@@ -184,6 +184,45 @@ pub fn scc<N, E, Ty, Ix>(g: &Graph<N, E, Ty, Ix>) -> Vec<Vec<NodeIndex<Ix>>>
     sccs
 }
 
+/// Condense every strongly connected component into a single node and return the result.
+///
+/// If `make_acyclic` is true, self-loops and multi edges are ignored, guaranteeing that
+/// the output is acyclic.
+pub fn condensation<N, E, Ty, Ix>(g: Graph<N, E, Ty, Ix>, make_acyclic: bool) -> Graph<Vec<N>, E, Ty, Ix>
```

Comment:
Yes, edges inside a component are turned into self-loops. I'm not sure even a two-argument signature would be needed for actual use cases. If you think that self-loops are generally uninteresting, we can just discard them and rename `make_acyclic` to `ignore_multi_edges`.